### DISCUSSION TITLE
Tracking the evolution of results

### DISCUSSION THREAD

**Discussion by tuscland** [probabl-ai/skore#576] [Ideas]
As a data scientist, I want to record the history of items I save in skore, in order to be able to access and visualize the evolution of those results.

Let's discuss this user story and see what solutions we might come up with.

**Comment by tuscland** [comment]
With @augustebaum we discussed how we might offer this feature.

First, skore should record items with as many meta-data as possible, including the datetime, as explained in #120.

With this, we should allow putting items with the same name as many times as desired. Tracking the evolution of an item is thus a matter of querying the database.

By default, viewing a single item shows its latest instance.

We could also offer a way to display items in more flexibles ways. The user click a `+` button in the current view, selects an item to plot, and fill out some parameters to display a dynamically generated plot or chart.

For example, you could choose to display a scatter plot of an item called `temperature` choosing its value on the `X` axis and it's timestamp on the `Y` axis. Or if you have a primitive item which is a dictionary with `a` and `b` keys that are numbers, you could plot all the instances of this items on a scatter plot.

You could also filter by git commit, or tag, or whatever metadata that has been recorded…

**Comment by rouk1** [comment]
We may also store different version of a plot and consult them side by side to play a 7 differences game.

**Comment by anasstarfaoui** [comment]
So, we would want to be able to create some elements directly inside the UI right ?

**Comment by tuscland** [comment]
> So, we would want to be able to create some elements directly inside the UI right ?

Yes, and programmatically as well.

**Comment by anasstarfaoui** [comment]
Relation with this research ticket: https://github.com/probabl-ai/skore/issues/477
